script/precompile_tasks

```diff
@@ -3,7 +3,15 @@
 # Precompile tasks if not in production and not in CI
 # Also allow skipping precompilation with an ENV var
 if [ "$LUCKY_ENV" != "production" ] && [ -z "$CI" ] && [ -z "$SKIP_LUCKY_TASK_PRECOMPILATION" ]; then
-    shards build
+    # This is to prevent Lucky's development dependencies being installed
+    # from an external source like running `shards install` from a Lucky app.
+    # Install shards and build shard targets.
+    if [ -z "$BUILD_WITHOUT_DEVELOPMENT" ]; then
+        shards build
+    else
+        shards build --without-development
+    fi
```

Comment:
Since the typical use case here would be to build without development, should this `if` be reversed so that the flag doesn't need to be passed in unless you actually want the ancillary binaries?

Do we even need the flag here? Assuming that I actually want the ameba binary I can add that to my own shards.build stanza for the project.

Reply:
My bash skills suck 😅  it should probably be flipped

@edwardloveall and I were talking about this, and we figured if you're working on Lucky directly, maybe there's a time you want to run that script with any dev deps included? I think the 99% case though would be without.